collated/main/A0153440R.md

```diff
@@ -0,0 +1,806 @@
+# A0153440R
+###### \java\jym\manager\logic\commands\AddCommand.java
+``` java
+    public AddCommand(String description, Object ... objects) throws IllegalValueException{
+    	if(objects.length > 3){//f**k in this case - this should never happen b/c we control parser. max args is date(s), location, priority
+    		throw new IllegalArgumentException();
+    	}
+    	boolean isTask = true; //by default
+    	for(Object o : objects){
+    		if(o instanceof List){
+    			List<LocalDateTime> d = (List<LocalDateTime>)o;
+    			if(d.size() > 1)
+    				isTask = false;
+    			else
+    				o = d.get(0);
+    		}
+    	}
+//      final Set<Tag> tagSet = new HashSet<>();
+//      for (String tagName : tags) {
+//          tagSet.add(new Tag(tagName));
+//      }
+ //   	if(isTask){
+        	this.toAdd = new Task(new Description(description), objects);
+  //  	} else {
+   // 		this.toAdd = new Event(new Description(description), objects);
+ //   	}
+    }
+    public AddCommand(String description, Set<String> tags) throws IllegalValueException
+    {
+        final Set<Tag> tagSet = new HashSet<>();
+        for (String tagName : tags) {
+            tagSet.add(new Tag(tagName));
+        }
+        this.toAdd = new Task(
+                new Description(description),
+                new UniqueTagList(tagSet)
+        );
+    }
+    public AddCommand(String description, LocalDateTime deadline, Set<String> tags) throws IllegalValueException
+    {
+        final Set<Tag> tagSet = new HashSet<>();
+        for (String tagName : tags) {
+            tagSet.add(new Tag(tagName));
+        }
+        this.toAdd = new Task(
+                new Description(description),
+                deadline,
+                new UniqueTagList(tagSet)
+        );
+    }
+    public AddCommand(String description, LocalDateTime deadline) throws IllegalValueException
+    {
+     
+        this.toAdd = new Task(
+                new Description(description),
+                deadline
+        );
+    }
+```
+###### \java\jym\manager\logic\commands\EditCommand.java
+``` java
+
+public class EditCommand extends Command {
+    public static final String COMMAND_WORD = "update";
```

Comment:
Is `public` the best access modifier for `COMMAND_WORD` ? As a rule of thumb start all data members as private, and depending on their usage promote them to protected, and if absolutely necessary, to public.